src/JustSaying/JustSayingFluently.cs

```diff
@@ -469,16 +465,20 @@ public IInterrogationResponse WhatDoIHave()
             return (Bus as IAmJustInterrogating)?.WhatDoIHave();
         }
 
-        public IMayWantOptionalSettings WithNamingStrategy(Func<INamingStrategy> busNamingStrategy)
+        public IMayWantOptionalSettings WithAwsClientFactory(Func<IAwsClientFactory> awsClientFactory)
         {
-            _busNamingStrategyFunc = busNamingStrategy;
+            _awsClientFactoryProxy.SetAwsClientFactory(awsClientFactory);
             return this;
         }
 
-        public IMayWantOptionalSettings WithAwsClientFactory(Func<IAwsClientFactory> awsClientFactory)
+        private static string GetOrUseDefaultTopicName<T>(string topicName)
         {
-            _awsClientFactoryProxy.SetAwsClientFactory(awsClientFactory);
-            return this;
+            return string.IsNullOrWhiteSpace(topicName) ? typeof(T).ToTopicName() : topicName;
+        }
+
+        private static string GetOrUseDefaultQueueName<T>(string queueName)
+        {
+            return string.IsNullOrWhiteSpace(queueName) ? typeof(T).ToTopicName() : queueName;
```

Comment:
`typeoof(T).ToQueueName()`?

Reply:
That would make sense however I just wanted to mirror what was in the soon-to-be-possibly-removed [DefaultNamingStrategy](https://github.com/justeat/JustSaying/blob/master/src/JustSaying/DefaultNamingStrategy.cs#L20)

Reply:
I figured but maybe as you are changing it, it might be worth having a `ToQueueName` extension name too. I am not that fussed about it though.

Reply:
Sure, I'll make that change

Reply:
Added `ToQueueName()` and ensuring the resource name lengths are respected for queues and topics